client-misk-hibernate/src/main/kotlin/app/cash/backfila/client/misk/hibernate/internal/BoundingRangeStrategy.kt

```diff
@@ -181,7 +181,15 @@ private fun <E : DbEntity<E>, Pkey : Any> selectMaxBound(
 private fun <E : DbEntity<E>, Pkey : Any> schemaAndTable(backfill: HibernateBackfill<E, Pkey, *>): String {
   val tableAnnotation = backfill.entityClass.java.getAnnotation(Table::class.java)
   val schema = tableAnnotation.schema
-  val table = tableAnnotation.name
+  val table = tableAnnotation.name.ifEmpty {
+      // We require a table name to encourage best practices. An entity is singular while your table
+      // is plural. Additionally, you probably want to name your classes in a way to make them
+      // obviously entity classes. Such as prefixing all entities with `DB`.
+      //
+      // You are welcome to create a copy of the bounding range strategy limited to your service
+      // if you absolutely cannot have table name.
+      error("Entity class ${backfill.entityClass.simpleName} is missing Table name.")
```

Comment:
Should this comment just be part of the error message?

```suggestion
    error(
      """
      Entity class ${backfill.entityClass.simpleName} is missing Table name.
      
      We require a table name to encourage best practices. An entity is singular while your table
      is plural. Additionally, you probably want to name your classes in a way to make them
      obviously entity classes. Such as prefixing all entities with `DB`.

      You are welcome to create a copy of the bounding range strategy limited to your service
      if you absolutely cannot have table name annotation on your entity class.
      """.trimIndent()
    )
```